### PR TITLE
Fix wrong separated components in messageInputBar

### LINKED ISSUE #569: Hybrid multi-language message be seperated
## General Information

* MessageKit Version:0.13.1

* iOS Version(s): 11.2

* Swift Version:4.0

* Devices/Simulators:iPhone6sPlus

* Reproducible in ChatExample? (Yes/No):Yes

##  What happened?
When new message contains hybrid multi-language like Chinese + English, for example，"你好Jane", the message will be seperated in two messages: "你好" and "Jane"

## What did you expect to happen?
The sent message should keep original one message.

### PR DESCRIPTION
**Pull request by zhongwuzw**
Fix #569

### CODE REVIEW

**Review by SD10 — COMMENTED** [review]
I'm good with this but I got lost on the second half of the logic. It may be good to wait and have @nathantannar4 review this as he is the original author.

File: Sources/Views/InputTextView.swift [hunk 1]
```diff
@@ -303,26 +303,50 @@ open class InputTextView: UITextView {
...
+                    attachments.append((range, image))
+                } else if let image = attachment.image(forBounds: attachment.bounds,
+                                                       textContainer: nil,
+                                                       characterIndex: range.location) {
+                    attachments.append((range,image))
+                }
+            }
+        }
+        
+        var curLocation = 0
```
Comment (SD10) [inline Sources/Views/InputTextView.swift:321]: @zhongwuzw Can you help me understand the code from here? I just want to understand before we merge if Nathan doesn't have time to look at this one

Comment (zhongwuzw) [reply Sources/Views/InputTextView.swift:321]: @SD10 , First we should enumerate all `.attachment`, get its image and range, then we can split text and image based on the range.

File: Sources/Views/InputTextView.swift [hunk 2]
```diff
@@ -303,26 +303,50 @@ open class InputTextView: UITextView {
...
+        if attachments.count == 0 {
```
Comment (zhongwuzw) [inline Sources/Views/InputTextView.swift:322]: @SD10 If no images, just text.

File: Sources/Views/InputTextView.swift [hunk 3]
```diff
@@ -303,26 +303,50 @@ open class InputTextView: UITextView {
...
+            if !text.isEmpty {
+                components.append(text)
+            }
+        }
+        else {
+            attachments.forEach { (attachment) in
+                let (range, image) = attachment
+                if curLocation < range.location {
+                    let textRange = NSMakeRange(curLocation, range.location)
+                    let text = attributedText.attributedSubstring(from: textRange).string.trimmingCharacters(in: .whitespacesAndNewlines)
```
Comment (zhongwuzw) [inline Sources/Views/InputTextView.swift:333]: Add text and image to `components`

File: Sources/Views/InputTextView.swift [hunk 4]
```diff
@@ -303,26 +303,50 @@ open class InputTextView: UITextView {
...
                 }
-            } else {
-                let stringValue = attributedText.attributedSubstring(from: range).string.trimmingCharacters(in: .whitespacesAndNewlines)
-                if !stringValue.isEmpty {
-                    components.append(stringValue)
+                
+                curLocation = range.location + range.length
+                components.append(image)
+            }
+            if curLocation < length - 1  {
```
Comment (zhongwuzw) [inline Sources/Views/InputTextView.swift:342]: After last `image`, maybe still has text, so we should append it to `components`.

Comment (SD10) [reply Sources/Views/InputTextView.swift:342]: Why is `length - 1` here? Sorry 😓

Comment (zhongwuzw) [reply Sources/Views/InputTextView.swift:342]: :smile: It's because we add a new line after image, so we decrease 1 to prevent unnecessary calculation.

**Review by SD10 — APPROVED** [review]
@zhongwuzw Thanks for walking me through this 👍

**Pull request merged**